### PR TITLE
Add more e2e auto completion tests.

### PR DESCRIPTION
**Pull request by YoyinZyc**
added tests and fixed some small bugs during this process.
@logicalhan @DirectXMan12

### CODE REVIEW

File: promq/autocomplete/earley/earley.go [hunk 1]
```diff
@@ -206,7 +208,7 @@ func (p *Earley) GetSuggestedTokenType(tokens Tokens) (types []ContextualToken)
 			// No further parsing needed if input tokens is exactly the previous input
 		}
 	}
-
+	//fmt.Println(p.chart.String())
```
Comment (logicalhan) [inline promq/autocomplete/earley/earley.go]: delete?

File: promq/autocomplete/earley/completer.go [hunk 2]
```diff
@@ -107,86 +107,40 @@ func (c *promQLCompleter) GenerateSuggestions(query string, pos int) []autocompl
...
 			}
-		case FUNCTION_VECTOR_ID:
-			for _, ao := range autocomplete.FilterPrefix(sets.StringKeySet(vectorFunctions), autocompletePrefix, false).List() {
-				newMatch := NewPartialMatch(ao, "label-match", vectorFunctions[ao])
-				matches = append(matches, newMatch)
-			}
-		case FUNCTION_SCALAR_ID:
-			for _, ao := range autocomplete.FilterPrefix(sets.StringKeySet(scalarFunctions), autocompletePrefix, false).List() {
-				newMatch := NewPartialMatch(ao, "label-match", scalarFunctions[ao])
+		case newStringSet(tokenTypes...).Has(string(s.TokenType)):
```
Comment (logicalhan) [inline promq/autocomplete/earley/completer.go]: isn't newStringSet(tokenTypes...) basically a constant? shouldn't we just do this once and cache it?

Comment (YoyinZyc) [reply promq/autocomplete/earley/completer.go]: Oh yes. Added var `tokenTypeStringSet` in promql.go

File: promq/autocomplete/earley/earley.go (hunk 1 above)
Comment (YoyinZyc) [reply promq/autocomplete/earley/earley.go]: Deleted.

**Review by logicalhan — APPROVED** [review]
/lgtm

**Pull request merged**